datadog_checks_dev/datadog_checks/dev/tooling/manifest_validator/common/validator.py

```diff
@@ -0,0 +1,273 @@
+    METRIC_PATH = {"v1": "/metric_to_check", "v2": "/assets/integration/metrics/check"}
+    PRICING_PATH = {"v1": "/pricing", "v2": "/pricing"}
+
+    def validate(self, check_name, decoded, _):
+        if not self.should_validate() or check_name == 'snmp' or check_name == 'moogsoft':
+            return
+
+        metadata_path = self.METADATA_PATH[self.version]
+        metadata_in_manifest = decoded.get_path(metadata_path)
+
+        # metric_to_check
+        metric_path = self.METRIC_PATH[self.version]
+        metric_to_check = decoded.get_path(metric_path)
+
+        pricing = decoded.get_path("/pricing") or []
+
+        if metric_to_check:
+            metrics_to_check = metric_to_check if isinstance(metric_to_check, list) else [metric_to_check]
+            if any(p.get('metric') in metrics_to_check for p in pricing):
+                return
+            for metric in metrics_to_check:
+                metric_integration_check_name = check_name
+                # snmp vendor specific integrations define metric_to_check
+                # with metrics from `snmp` integration
+                if check_name.startswith('snmp_') and not metadata_in_manifest:
+                    metric_integration_check_name = 'snmp'
+                if (
+                    not is_metric_in_metadata_file(metric, metric_integration_check_name)
+                    and metric not in self.METRIC_TO_CHECK_EXCLUDE_LIST
+                ):
+                    self.fail(f'  metric_to_check not in metadata.csv: {metric!r}')
+
+        elif metadata_in_manifest:
+            # if we have a metadata.csv file but no `metric_to_check` raise an error
+            metadata_file = get_metadata_file(check_name)
+            if os.path.isfile(metadata_file):
+                for _, row in read_metadata_rows(metadata_file):
+                    # there are cases of metadata.csv files with just a header but no metrics
+                    if row:
+                        self.fail('  metric_to_check not included in manifest.json')
+
+
+class IsPublicValidator(BaseManifestValidator):
+    PUBLIC_PATH = {"v1": "/is_public", "v2": "/display_on_public_website"}
+
+    def validate(self, check_name, decoded, fix):
+        correct_is_public = True
+        path = self.PUBLIC_PATH[self.version]
+        is_public = decoded.get_path(path)
+        if not isinstance(is_public, bool):
+            output = '  required boolean: is_public'
+
+            if fix:
+                decoded.set_path(path, correct_is_public)
+                self.fix(output, f'  new `is_public`: {correct_is_public}')
+            else:
+                self.fail(output)
+
+
+class ImmutableAttributesValidator(BaseManifestValidator):
+    """Ensure attributes haven't changed
+    Skip if the manifest is a new file (i.e. new integration)
+    """
+
+    FIELDS_NOT_ALLOWED_TO_CHANGE = {
+        "v1": ("integration_id", "display_name", "guid"),
+        "v2": ("id", "source_type_name", "app_id"),
```

Comment:
I think these v2 versions need to be `assets/integration/id`, ``assets/integration/source_type_name`, and `app_id`

Reply:
The `content_changed` function below just returns git blocks which are what are being tested - so we wouldn't be able to fully trace a path like `assets/integration/id`.  It does mean we could end up catching other id fields in this case though ..